samples/tutorial/Tutorial5.md

```diff
@@ -0,0 +1,612 @@
+# Step 5
+
+_Unit and Integration Testing Workflows_
+
+## Setup
+
+To follow this tutorial, launch Android Studio and open this folder (`samples/tutorial`).
+
+Start from the implementation of `tutorial-4-complete` if you're skipping ahead.
+
+## Testing
+
+`Workflow`s being easily testable was a design requirement. It is essential to building scalable, reliable software.
+
+The `workflow-testing` library is provided to allow easy unit and integration testing. For this tutorial, we'll use the `kotlin-test` library to define tests and assertions, but feel free to use your favorite testing or assertion library instead – `workflow-testing` doesn't care.
+
+## Unit Tests: `WorkflowAction`s
+
+A `WorkflowAction`'s `apply` function is effectively a reducer. Given a current state and action, it returns a new state (and optionally an output). Because an `apply` function should almost always be a "pure" function, it is a great candidate for unit testing.
+
+The `applyTo` extension function is provided to facilitate writing unit tests against actions.
+
+### applyTo
```

Comment:
Totally did not think of this method as a testing aid. Love it.

Reply:
Why is it `@ExperimentalWorkflowApi`?

Reply:
I don't know. I'm down to remove that.